fix: Icon updates and equipping items via the actionbar

andreoam opened the pull request:
# Description

- Updated icons

- Fixed the borders of rare icons when vbot is active
- Updated action bar
Now the actionbar equips items, inside closed backpacks. Ideal for SSA macros or quick equipping with hotkeys
- Updated the open gamestore button in the prey window

### Before
https://github.com/user-attachments/assets/6a5fd80d-b8f0-4367-800c-d76da2f0fd7d

### After
https://github.com/user-attachments/assets/78e95551-a52a-468a-afe0-2665f40247aa

andreoam committed:
Update game_actionbar.lua

fix: equip action bar item with closed backpack

andreoam committed:
up

andreoam committed:
Prey

Adicionado ícone igual cliente cip com link direto para gamestore

andreoam committed:
Update shop.png

andreoam committed:
update icons

andreoam committed:
up

Vysius commented on mods/game_bot/default_configs/vBot_4.8/vBot/analyzer.lua:
```diff
@@ -1065,7 +1065,7 @@ end
 
 local function setFrames()
-  if g_game.getFeature(GameColorizedLootValue) then return end
+  if g_game.getFeature(121) then return end
```
Do not use magic number (121) here, use a constant.

kokekanon replied:
add https://github.com/mehah/otclient/blob/main/mods/game_bot/functions/const.lua

`context.GameColorizedLootValue = GameColorizedLootValue`

Vysius commented on modules/game_actionbar/game_actionbar.lua:
```diff
@@ -667,7 +678,7 @@ function setupHotkeys()
                     modules.game_hotkeys.executeHotkeyItem(HOTKEY_USEONSELF, slot.itemId, slot.subType)
                 elseif slot.useType == 'equip' then
-                    local item = g_game.findPlayerItem(slot.itemId, -1, slot.getTier)
+                    local item = Item.create(slot.itemId)
```
Why create instead getting from player?

andreoam replied:
obter do player so estava equipando os itens em backpacks abertas, Utilizei a mesma logica do otcv8

Vysius replied:
English please

Vysius reviewed (changes requested):
I dont really get the changes in actionbar. Someone else review too

andreoam committed:
Merge branch 'mehah:main' into main

kokekanon commented:
modify the description in English , so that we all understand , and modify the title with the prefix "fix:"

kokekanon commented:
in modules` /game_prey/prey.otui` there are several calls to `modules.game_shop.show()`

should be `@onClick: modules.game_mainpanel.toggleStore()`

![image](https://github.com/user-attachments/assets/7b0ff2be-cfc4-4d17-8681-614ca767e34c)

andreoam committed:
updating suggestions

update prey.otui
add constant GameColorizedLootValue

andreoam commented:
updated suggestions

added video of actionbar functionality

kokekanon commented:
> updated suggestions
> 
> added video of actionbar functionality

try with 2 equal items with tier and without tier

this fix that :
![image](https://github.com/user-attachments/assets/47c5c1f9-532b-408c-ad39-49deb79122dd)

andreoam commented:
> > sugestões atualizadas
> > adicionado vídeo da funcionalidade da barra de ação
> 
> tente com 2 itens iguais com nível e sem nível
> 
> esta correção que: ![imagem](https://private-user-images.githubusercontent.com/114332266/423243618-47c5c1f9-532b-408c-ad39-49deb79122dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIxNTM3NjYsIm5iZiI6MTc0MjE1MzQ2NiwicGF0aCI6Ii8xMTQzMzIyNjYvNDIzMjQzNjE4LTQ3YzVjMWY5LTUzMmItNDA4Yy1hZDM5LTQ5ZGViNzkxMjJkZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxNlQxOTMxMDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYjk3OTM4N2I2Y2ExNGFhMzdkNGRhYWUwZGFiZjQxNjY4NjIwZmRjMTViYzUxMDcyNjdkZDU2YzBjYmM4MTRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._YbkQEztVRnguWEomofG515OZVBi6zkSYbubB2CSO8M)

what does it do?

kokekanon commented:
> what does it do?

I see you use canary

/i 3388,1
/i 3388,1,5
add action bar

![image](https://github.com/user-attachments/assets/91ba4873-324c-4711-984b-8112cb0f5f23)

when you click to equip...
currently difference between item tier ? no

andreoam committed:
update actionbar

added setTier to luafunctions.cpp and updated game_actionbar.lua to use setTier

andreoam commented:
> > o que ele faz?
> 
> Vejo que você usa canário
> 
> /i 3388,1 /i 3388,1,5 adicionar barra de ação
> 
> ![imagem](https://private-user-images.githubusercontent.com/114332266/423245030-91ba4873-324c-4711-984b-8112cb0f5f23.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIxNTQ2OTEsIm5iZiI6MTc0MjE1NDM5MSwicGF0aCI6Ii8xMTQzMzIyNjYvNDIzMjQ1MDMwLTkxYmE0ODczLTMyNGMtNDcxMS05ODRiLTgxMTJjYjBmNWYyMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxNlQxOTQ2MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NWE3ZDYxNDUxMjhhOTk1NzJmMzM5ZTY1YjM5MmQ5OGJmN2IyNzU3YzZlMjU2MTM3MTEyMDYwYzg4YjIyN2UzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.7v6Gy5dfN9MmAoy181Cb4-9mAvjrw6fr0U-YsHdlcjQ)
> 
> quando você clica para equipar... atualmente diferença entre o nível do item? não

okay

updated

kokekanon commented on src/client/luafunctions.cpp:
```diff
@@ -710,7 +710,7 @@ void Client::registerLuaFunctions()
     g_lua.bindClassMemberFunction<Item>("setCount", &Item::setCount);
     g_lua.bindClassMemberFunction<Item>("setTooltip", &Item::setTooltip);
-
+    g_lua.bindClassMemberFunction<Item>("setTier", &Item::setTier);
```
```suggestion
    g_lua.bindClassMemberFunction<Item>("setTier", &Item::setTier);

```

kokekanon commented on modules/game_actionbar/game_actionbar.lua:
```diff
@@ -1135,4 +1148,4 @@ end
 function getPanelActionbar()
     return actionBar
-end
+end
```
```suggestion
end

```

kokekanon commented on mods/game_bot/functions/const.lua:
```diff
@@ -23,3 +23,4 @@ context.InventorySlotAmmo = 10
 context.InventorySlotPurse = 11
 context.InventorySlotFirst = 1
 context.InventorySlotLast = 10
+context.GameColorizedLootValue = GameColorizedLootValue
```
```suggestion

context.GameColorizedLootValue = GameColorizedLootValue

```

andreoam committed:
Merge branch 'mehah:main' into main

andreoam committed:
Update src/client/luafunctions.cpp

Co-authored-by: kokekanon <114332266+kokekanon@users.noreply.github.com>

andreoam committed:
Update mods/game_bot/functions/const.lua

Co-authored-by: kokekanon <114332266+kokekanon@users.noreply.github.com>
